packages/alfa-frontier/src/frontier.ts

```diff
@@ -0,0 +1,370 @@
+import { Equatable } from "@siteimprove/alfa-equatable";
+import { Serializable } from "@siteimprove/alfa-json";
+import { Option, None } from "@siteimprove/alfa-option";
+import { Predicate } from "@siteimprove/alfa-predicate";
+
+import * as json from "@siteimprove/alfa-json";
+
+const { equals, property } = Predicate;
+
+export class Frontier implements Equatable, Serializable {
+  public static of(
+    scope: string | URL,
```

Comment:
I can see the use case where the scope is more than one domain (URL). E.g. wantirg to crawl `example1.com` and `example.com`.

Reply:
That seems like a quick way to explode the scope though, no pun intended. For example, one might want to use a single frontier to be scoped to subdomains as well, so then we need to support glob patterns like `https://*.example.com`.

I'd prefer to keep it simple and limit every frontier to a single URL prefix as its scope. It multiple distinct scopes are needed, multiple frontiers must be used.